c10/util/Bitset.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+#include <c10/util/C++17.h>
+#include <c10/util/Optional.h>
+#include <iostream>
+
+namespace c10 {
+namespace utils {
+
+/**
+ * This is a simple bitset class with sizeof(long long int) bits.
+ * You can set bits, unset bits, query bits by index,
+ * and query for the first set bit.
+ * Before using this class, please also take a look at std::bitset,
+ * which has more functionality and is more generic. It is probably
+ * a better fit for your use case. The sole reason for c10::utils::bitset
+ * to exist is that std::bitset misses a find_first_set() method.
+ */
+struct bitset final {
+ public:
+  static constexpr size_t NUM_BITS = 8 * sizeof(long long int);
+
+  constexpr bitset() noexcept : bitset_(0) {}
+  constexpr bitset(const bitset&) noexcept = default;
+  constexpr bitset(bitset&&) noexcept = default;
+  constexpr bitset& operator=(const bitset&) noexcept = default;
+  constexpr bitset& operator=(bitset&&) noexcept = default;
+
+  constexpr void set(size_t index) noexcept {
+    bitset_ |= (static_cast<long long int>(1) << index);
+  }
+
+  constexpr void unset(size_t index) noexcept {
+    bitset_ &= ~(static_cast<long long int>(1) << index);
+  }
+
+  constexpr bool get(size_t index) const noexcept {
+    return bitset_ & (static_cast<long long int>(1) << index);
+  }
+
+  // Call the given functor with the index of each bit that is set
+  template <class Func>
+  void for_each_set_bit(Func&& func) const {
+    bitset cur = *this;
+    size_t index = ffsll(cur.bitset_);
```

Comment:
you can try to use __builtin_ffs - might be faster

Reply:
Why do you think it might be faster? Isn't ffsll internally just calling __builtin_ffsll?

Reply:
I wasn't able to find those calls in the assembly (does that maybe mean they're not inlined???) but the `getDispatchKeyBoxed` function generated exactly the same assembly with `__builtin_ffsll` than with `ffsll`

Reply:
That should be good then - I was posting very low-confidence comments to see whether perf regression comes from there. Given that we found the source of regression - we can disregard it.